src/aws/osml/tile_sever_test/processor_base.py

```diff
@@ -0,0 +1,50 @@
+#  Copyright 2024 Amazon.com, Inc. or its affiliates.
+
+import json
+import traceback
+from abc import ABC, abstractmethod
+from typing import Any, Dict
+
+from .utils.logger import logger
+
+
+class ProcessorBase(ABC):
+    """
+    A base class providing common success and failure message handling for processors.
+    """
+
+    @staticmethod
+    def success_message(message: str) -> Dict[str, Any]:
+        """
+        Returns a success message in the form of a dictionary, intended for an HTTP response.
+
+        :param message: The success message to send when complete.
+        :returns: A dictionary with 'statusCode' set to 200 and a 'body' containing a success message.
+        """
+        logger.info(message)
+        return {"statusCode": 200, "body": json.dumps(message)}
+
+    @staticmethod
+    def failure_message(e: Exception) -> Dict[str, Any]:
+        """
+        Returns an error message in the form of a dictionary, including a stack trace, intended for an HTTP response.
+
+        :param e: The exception that triggered the failure.
+        :returns: A dictionary with 'statusCode' set to 400 and a 'body' containing the error message and stack trace.
+        """
+        # Log the error and stack trace
+        stack_trace = traceback.format_exc()
+        logger.error(f"Error creating STAC items: {e}\nStack trace: {stack_trace}")
+
+        # Return the error message and stack trace in the response
+        error_response = {"message": str(e), "stack_trace": stack_trace.splitlines()}
+        return {"statusCode": 400, "body": json.dumps(error_response)}
```

Comment:
Failure/Fault class exceptions are usually 500s whereas Error class client exceptions are 400s. You sure you want 400 here?

Reply:
This is a good catch - this should probably be a 500 error.